Students/Pop Calin/First assignments/LargestPalindrom.java

```diff
@@ -0,0 +1,52 @@
+package assignments;
+
+public class LargestPalindrom 
+{
+	
+	
+		public static void main(String[] args)
+		{
+			
+			largestPalindrom(999,100);
+			largestPalindrom(9999,1000);
+		}	
+				
+		
+		public static boolean palindrom(int number)
+		{
+			int numberPalindrom = 0;
+			int numberOriginal = number;
+			
+			while (numberOriginal>0)
+			{
+				numberPalindrom = 10 * numberPalindrom + numberOriginal % 10;
+				numberOriginal /= 10;
+			}
+			
+			if (number == numberPalindrom)
+				  return true;
+			else
+				  return false;
+		}
+		
+		public static void largestPalindrom(int sup, int inf)
+		{
+
+			int maxI = 0;
+			int maxJ= 0;
+			int maxPalindrom = 0;
+			
+			for (int i=sup;i>=inf;i--)
+				for (int j=sup;j>=inf;j--)
+				{
+					if (palindrom(i*j) == true  && (i*j)> maxPalindrom)
```

Comment:
This is correct but a more efficient approach would be this: since you start from the largest 3-digit/4-digit numbers and you multiply them, it's natural that the first palindrom is also the maximum, right?

So you can keep only the first part of the condition and break both `for` loops when you enter on `if` branch